### PR TITLE
1020 - Update the code to add the aria-label property to the modus-modal's both buttons

### LINKED ISSUE #1020: [BUG] Modal Buttons missing accessibility attributes
**Describe the bug**
Modal Button (primary-button and secondary-button) are missing accessibility attributes

**To Reproduce**

Go to https://modus-web-components.trimble.com/?path=/docs/components-modal--default
Click on 'open modal'
The Close button should have attributes role="button" and aria-label="Close"

1. Go to https://modus-web-components.trimble.com/?path=/docs/components-modal--default
2. Click on 'open modal'
3. Scroll down to the popup model buttons (primary-button ("Sweet!") and secondary-button ("Save changes"))
4. Both buttons are missing aria-label

**Expected behavior**
The modus-modal's both buttons should have the aria-label property for the accessibility

**Screenshots**
image

### PR DESCRIPTION
**Pull request by hrgondaliya**
Update the code to add the aria-label property to the modus-modal's both buttons

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1020

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
We also need to change the properties for the Modal component on the storybook page.

File: stencil-workspace/src/components.d.ts [hunk 1]
```diff
@@ -423,10 +423,18 @@ export namespace Components {
...
+         */
+        "primaryButtonAreaLabel": string | null;
         /**
           * (optional) The modal's primary button text.
          */
         "primaryButtonText": string;
+        /**
+          * (optional) The modal's secondary button Area Label.
+         */
+        "secondaryButtonAreaLabel": string | null;
```
Comment (msankaran0712) [inline stencil-workspace/src/components.d.ts:437]: I believe it should read  `..AriaLabel`

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

**Comment by msankaran0712** [comment]
It would be nice to have something like https://shoelace.style/components/dialog?id=slots for the footer section as optional, what do you think? @ryan-henness-trimble

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 2]
```diff
@@ -23,6 +23,12 @@ export class ModusModal {
   /** (optional) The modal's z-index. */
   @Prop() zIndex = '1';
 
+  /** (optional) The modal's primary button Area label. */
+  @Prop() primaryButtonAreaLabel: string | null;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:27]: I believe it should read ..AriaLabel, also in the description text.

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 3]
```diff
@@ -23,6 +23,12 @@ export class ModusModal {
...
+
+  /** (optional) The modal's secondary button Area Label. */
+  @Prop() secondaryButtonAreaLabel: string | null;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:30]: I believe it is sufficient to use only `string` for prop type @ryan-henness-trimble please correct me if am wrong.

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx (hunk 3 above)
Comment (msankaran0712) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx]: I recommend keeping only `string`.

Comment (ryan-henness-trimble) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx]: If I remember correctly, the Storybook accessibility checks were not passing unless we included `null`

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 4]
```diff
@@ -23,6 +23,12 @@ export class ModusModal {
   /** (optional) The modal's z-index. */
   @Prop() zIndex = '1';
 
+  /** (optional) The modal's primary button Aria label. */
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:26]: Can we change `Aria label` to `aria-label` in the comments?

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 5]
```diff
@@ -23,6 +23,12 @@ export class ModusModal {
...
+  @Prop() primaryButtonAriaLabel: string | null;
+
+  /** (optional) The modal's secondary button Aria Label. */
+  @Prop() secondaryButtonAriaLabel: string | null;
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:30]: Let's alphasort these `@Prop()`s - that way they're next to their corresponding `ButtonText` prop

**Comment by ryan-henness-trimble** [comment]
> It would be nice to have something like https://shoelace.style/components/dialog?id=slots for the footer section as optional, any thoughts? @ryan-henness-trimble

It depends on the Modus spec, do we want to allow the user to add more than 3 buttons? Do we want them to be able to render elements in the footer other than buttons?

**Review by ryan-henness-trimble — APPROVED** [review]

**Comment by msankaran0712** [comment]
> > It would be nice to have something like https://shoelace.style/components/dialog?id=slots for the footer section as optional, any thoughts? @ryan-henness-trimble
> 
> It depends on how rigid we want to adhere to the Modus specs. Do we want to allow the dev to add more than two buttons? Do we want them to be able to render elements other than buttons in the footer?
> 
> If we want to give them full control, a slot sounds like a good option. But if we want to limit what can be rendered in the footer, we might want to consider another option
> 
> I would lean towards adding optional slots that will override the default buttons. More control seems better in this case

@ryan-henness-trimble I agree to have optional slots for buttons. It is true that users should not add buttons more than two, but providing a slot as an option will give users better control over the button configurations instead of adding those configs in the Modal's API, which in the future might become too much.

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx (hunk 3 above)
Comment (msankaran0712) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx]: hey, @ryan-henness-trimble @hrgondaliya I tested the storybook page by removing `null` from the properties and keeping only `string` as a type, seems like there is no issue with accessibility, so up to you to decide whether to remove `null` or keep it.

**Pull request merged**